targets/TARGET_STM/TARGET_STM32F0/i2c_device.c

```diff
@@ -78,7 +78,13 @@ uint32_t i2c_get_timing(I2CName i2c, int hz)
     }
 
     else {
+        /* If MBED_CONF_TARGET_I2C_TIMING_VALUE_ALGO assert is triggered. 
+        User needs to enable I2C_TIMING_VALUE_ALGO in target.json for specific target. 
+        Enabling this may impact performance*/
+        MBED_ASSERT(MBED_CONF_TARGET_I2C_TIMING_VALUE_ALGO);
```

Comment:
Shouldn't this be an error rather and print a message how to fix it?

Reply:
Mentioned as a comment what to do assert occurred. So I think this will be enough.